tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -43,10 +43,10 @@ struct spi_cs_control spi_cs = {
 #define BUF2_SIZE 36
 
 #if CONFIG_NOCACHE_MEMORY
-static const char tx_data[] = "0123456789abcdef\0";
+static const char tx_data[BUF_SIZE] = "0123456789abcdef\0";
```

Comment:
OK, so the bug here is that BUF_SIZE was, I guess, 16, and this is actually an 18 byte array (with... TWO nulls at the end to terminate!?).  And something was doing the copy over sizeof(tx_data) and not BUF_SIZE?

I don't doubt that this is a fix, but this seems like it was immensely fragile code to begin with.  But I guess it's a test, and accepting questionable code in tests isn't really a bad thing as it exposes edges in the core software, right?

Reply:
BUF_SIZE is actually 17, the issue here was the explicit NUL termination which, you said, meant that there were two NULs at the end of the string, making the size 18 instead of 17. I decided that it was better to explicitly declare the correct size rather than remove the explicit NUL termination and allow the compiler to implicitly NUL terminate the string to 17 bytes. 

As you say, it's just test code, so we're not as concerned about making the code as robust as possible. I'm notl concerned about what this code looks like, as long as it generates the right compiler output :-)